.github/workflows/vtk-pre-test.yml

```diff
@@ -0,0 +1,66 @@
+name: VTK Master Testing
+
+on:
+  push:
+    branches:
+      # To resolve issues on VTK master and test, use this branch name pattern
+      - 'maint/vtk-master*'
+  workflow_dispatch:
+  schedule:
+    - cron: "0 4 * * *"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  ALLOW_PLOTTING: true
+  SHELLOPTS: 'errexit:pipefail'
+
+jobs:
+  VTK-master:
+    name: Linux VTK Master Testing
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.8, 3.9, "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pip'
+          cache-dependency-path: requirements_test.txt
+
+      - uses: actions/cache@v3
+        with:
+          path: ~/.local/share/pyvista/examples
+          key: Examples-1-${{ hashFiles('*') }}
+          restore-keys: |
+            Examples-1-
+
+      - name: Install pyvista
+        run: pip install . --no-deps
+
+      - name: Install Testing Requirements
+        run: pip install -r requirements_test.txt
+
+      - name: Set up VTK with OSMesa
+        run: |
+          pip uninstall vtk -y
+          pip install vtk_osmesa --pre --no-cache --extra-index-url https://wheels.vtk.org
```

Comment:
It would be good to have one run that uses VTK-osmesa and another that uses Xvfb-and-standard-VTK. The failure modes could be different and the latter is a (much more?) widely used test case.

Reply:
I think I will put that off for another PR -- due to current time constraints and because I want to use the new OSMesa wheel throughout our testing suite.